### PR TITLE
Split CodeQuestion into a separate superclass.

### PR DESCRIPTION
**Pull request by davis68**
This represents only the separation of PythonCodeQuestion into a daughter class.  No operational changes should be observed.

### CODE REVIEW

**Comment by davis68** [comment]
```
======================================================================
FAIL: test_docker_container_ping_failure (tests.test_pages.test_code.RequestPythonRunWithRetriesTest) (case='Docker ping socket error with erron ECONNREFUSED')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\relate\tests\test_pages\test_code.py", line 1122, in test_docker_container_ping_failure
    self.assertEqual(res["exec_host"], fake_host_ip)
AssertionError: 'localhost' != '192.168.1.100'
- localhost
+ 192.168.1.100
```

That's the last error set on this PR.  It looks like Docker is failing to spawn remotely.  This version works locally with the `inducer` image.  Can you take a look?  @inducer

**Comment by davis68** [comment]
OK, looks good now.  The lingering problem was because I have CodeQuestion define the Docker image.  The failure tests didn't originate from that object though, so there was a fallthrough case I had failed to consider.  It looks like it works now locally and remotely for everything without changing surface functionality.  I.e., with this PR nothing changes outside of the CodeQuestion internals, and there are no changes to be made to configuration files or anything else.

Placate flake8 and onwards.

File: course/page/code.py [hunk 1]
```diff
@@ -435,52 +443,14 @@ class PythonCodeQuestion(PageBaseWithTitle, PageBaseWithValue):
...
-          feedback.check_list(name, ref, data, entry_type=None)
-
-          feedback.check_scalar(name, ref, data, accuracy_critical=True,
-              rtol=1e-5, atol=1e-8, report_success=True, report_failure=True)
-          # returns True if accurate
-
-          feedback.call_user(f, *args, **kwargs)
-          # Calls a user-supplied function and prints an appropriate
-          # feedback message in case of failure.
-
```
Comment (inducer) [inline course/page/code.py]: This stuff still seemed Python-specific, deleted.

**Comment by inducer** [comment]
Sorry for the delay. LGTM, will merge once CI passes.

**Comment by inducer** [comment]
Whoops. Totally forgot to look for this after CI passed. Will merge in master and look for CI.

**Comment by inducer** [comment]
Hmm. Doesn't look like you've allowed me to push. Could you merge in master?

**Comment by inducer** [comment]
Oh well. I'll bring it in on the past CI success.

**Comment by davis68** [comment]
See https://github.com/inducer/relate/issues/631#issuecomment-520938924

This branch should be considered defunct.  I orthogonalized the changes into three separate collections of changes.

**Pull request merged**